libdevice/cmake/modules/SYCLLibdevice.cmake

```
@@ -356,9 +356,17 @@ add_devicelibs(libsycl-itt-user-wrappers
   SRC itt_user_wrappers.cpp
   DEPENDENCIES ${itt_obj_deps})
 
-add_devicelibs(libsycl-crt
-  SRC crt_wrapper.cpp
-  DEPENDENCIES ${crt_obj_deps})
+if (MSVC)
+  add_devicelibs(libsycl-crt
+    SRC crt_wrapper.cpp
+    DEPENDENCIES ${crt_obj_deps})
+else()
+  add_devicelibs(libsycl-crt
+    SRC crt_wrapper.cpp
+    DEPENDENCIES ${crt_obj_deps}
+    EXTRA_OPTS -Wno-invalid-noreturn)
+endif()
+
```

Using `check_cxx_compiler_flag` should be a more robust solution:

```suggestion
check_cxx_compiler_flag(-Wno-invalid-noreturn
  HAS_NO_INVALID_RETURN_WARN_FLAG)
add_devicelibs(libsycl-crt
  SRC crt_wrapper.cpp
  DEPENDENCIES ${crt_obj_deps}
  EXTRA_OPTS
    $<$<BOOL:${HAS_NO_INVALID_RETURN_WARN_FLAG}>:-Wno-invalid-noreturn>
  )

```

Note that with this you should also add
```
include(CheckLinkerFlag)
```
to the top of this file.